release.py

```diff
@@ -0,0 +1,488 @@
+
+        string_contained = 'release/release-' + tag_name
+        release_list_doc_path = os.path.join(self.directory, 'docs/release.rst')
+        self._get_and_check_release_file(release_list_doc_path,
+                                         string_contained=string_contained)
+
+        print 'Release docs confirmed for tag %s' % tag_name
+
+    def confirm_all_release_file_changes(self, tag_name):
+        self._confirm_version_changed(tag_name)
+        self._confirm_release_docs_format(tag_name)
+
+
+class GithubReleaser:
+    def __init__(self, directory, github_api_directory):
+        self.directory = directory
+        self.github_api_directory = github_api_directory
+        self.release_fetcher = ReleaseFetcher(directory, github_api_directory)
+        self.release_validator = ReleaseValidator(directory)
+        self.username = None
+        self.password = None
+        self.tag_name = None
+        self.release_name = None
+        self.target_commitish = None
+        self.name = None
+        self.body = None
+        self.is_draft = 'false'
+        self.is_prerelease = 'false'
+
+    def _user_set_username(self):
+        self.username = raw_input('Please input your Github username: ')
+
+    def _user_set_password(self):
+        self.password = getpass.getpass("Enter password for '%s': " % self.username)
+
+    def _get_authorization_string(self):
+        self._user_set_username()
+        self._user_set_password()
+        return base64.standard_b64encode('%s:%s' % (self.username, self.password))
+
+    def _check_and_set_release_fields(self):
+        """
+        This functions checks that files have been added and/or modified for the release.
+        It sets the fields necessary to release to Github.
+        """
+        self.target_commitish = self.release_fetcher.get_and_check_target_commitish()
+        self.tag_name = self.release_fetcher.get_and_check_tag()
+        self.release_validator.confirm_all_release_file_changes(self.tag_name)
+        self.body = self.release_fetcher.get_body_from_release_notes(self.tag_name)
+        self.body = self._make_multiline_string_json_friendly(self.body)
+        self.release_name = 'Release ' + self.tag_name
+
+    @staticmethod
+    def _make_multiline_string_json_friendly(multiline_string):
+        return multiline_string.replace('\n', '\\n')
+
+    def _build_json_post_contents(self):
+        return '{"tag_name": "%s", "target_commitish": "%s", "name": "%s", "body": "%s",' \
+               ' "draft": %s, "prerelease": %s}' \
+               % (self.tag_name, self.target_commitish, self.release_name,
+                  self.body, self.is_draft, self.is_prerelease)
+
+    @staticmethod
+    def _successful_post_request(status):
+        if status == 201:
+            return True
+        else:
+            return False
+
+    def _send_post_request(self, url, data, authorization_string, content_type, content_length):
```

Comment:
The GET/POST stuff probably belongs in a new module under `prestoadmin/util`. It's more general than just talking to github.